### ISSUE TITLE
Handling table definitions/CDM versioning changes

### ISSUE THREAD

**Issue by jbadger3** [DavidPageGroup/cdm-data#5]
@afbarnard 

Here is a cut an past from our initial discussion on how to handle data files that differ in format with successive mcrf data dumps.

"I managed to get the indexing to finish on the emr database and have now just realized the table structure of the SNOMED files is different from what we got last time.  (figures right?)  Essentially there are additional columns that both preserve the original data (ICD code, etc.) as well as provide mappings to standard concepts.  I have begun fixing things by making a copy of tables_emr.sqlite.sql and calling it tables_emr.snomed.sqlite.sql, but am looking for input on how you want things corrected for the github repo.  

My thoughts are we could 
1.  change the table definitions in tables_emr.sqlite.sql, assuming most folks will want to use the standardized data…which is also likely how Marshfield will be sending updates (a guess at this point)

or 

2.  have multiple versions of the table definitions to accommodate different versions of the source files.

My vote is number 1 and allowing for legacy support via the github repo.  The newer db will still have the old data for those wishing to use the original source values (ICD-9, etc).  The new table definitions will also more closely match the CDM specification from OHDSI.

Let me know your thoughts."

Response

"Overall, I would tend to having the tables follow the CDM and / or be compatible with the CDM such that any method developed for the CDM will work. But I'll have to look at the actual differences."

**Comment by jbadger3** [comment]
After thinking things over I think I am on board with your initial thoughts.  I think that our tables should follow the CDM specification from OMOP.  One issue that I see is that the specification has both required and optional fields for each table.  With the newest data dump that we have it includes more of the optional fields.  Perhaps we should have some upfront field checking/sqlite generation so we can adapt to whatever fields are included in the data files?

**Comment by afbarnard** [comment]
I haven't looked at this yet. Is it just that the SQLite SQL files don't specify all of the CDM columns?

**Comment by jbadger3** [comment]
Yes, that's the main problem.  I am also concerned that having hard coded sql leaves the code easily broken by OMOP model updates / Marshfield model updates.  I'm wondering if we should have a master OMOP CDM specification that we cross check with the Marshfield headers, and follow that with generating a .sql file based on the fields that are found.  Not sure how stable the table definitions will be in the next few years, but just a thought.

**Comment by jbadger3** [comment]
Not that I have seen so far.  The only deviation I have seen so far is with using the field study_id, which you have taken care of, and inclusion of additional fields.  

Your probably right that generating the SQL might be overkill.   With that said, I have edited versions of the SQL files I used for generating the current DB.  I'm not sure on the best naming scheme for multiple files.  In this case I don't think the new MCRF data follows a different version of the OMOP CDM, but rather they have included optional fields from the schema, so what would we want for additional tags?

**Comment by afbarnard** [comment]
If you are just adding additional fields specified in the CDM, then I would just update the files without changing their names. Perhaps you can indicate in a comment on each additional field that it was introduced with the 20180705 version of the data. Then a person who wants to rebuild the DB from the older data would only need to comment out some lines and those lines would be labeled. If you still think separate files are warranted, then we could tag them with the data date.
